cairo/tests/utils/args_gen.py

```diff
@@ -496,19 +511,39 @@ def _gen_arg(
             arg_type = Mapping[get_args(arg_type)[0], bool]
 
         data = {
-            _gen_arg(dict_manager, segments, get_args(arg_type)[0], k): _gen_arg(
-                dict_manager, segments, get_args(arg_type)[1], v
-            )
+            _gen_arg(
+                dict_manager,
+                segments,
+                get_args(arg_type)[0],
+                k,
+                hash_mode=hash_mode is not False,
+            ): _gen_arg(dict_manager, segments, get_args(arg_type)[1], v)
             for k, v in arg.items()
         }
+
         if isinstance_with_generic(arg, defaultdict):
             data = defaultdict(arg.default_factory, data)
 
         # This is required for tests where we read data from DictAccess segments while no dict method has been used.
         # Equivalent to doing an initial dict_read of all keys.
-        initial_data = flatten([(k, v, v) for k, v in data.items()])
+        # We only hash keys if they're in tuples.
+        initial_data = flatten(
+            [
+                (
+                    (
+                        poseidon_hash_many(tuple(k))
```

Comment:
```suggestion
                        poseidon_hash_many(k)
```
or 
```suggestion
                        poseidon_hash_many(list(k))
```
I think this is not necessary, it will be transformed in to list in the end 
https://github.com/starkware-libs/cairo-lang/blob/8e11b8cc65ae1d0959328b1b4a40b92df8b58595/src/starkware/cairo/common/poseidon_hash.py#L77